src/utils/combineReducers.js

```diff
@@ -100,15 +100,19 @@ export default function combineReducers(reducers) {
   var stateShapeVerified;
 
   return function combination(state = defaultState, action) {
-    var dirty = false;
-    var finalState = finalReducers.map((reducer, key) => {
-      var oldState = state.get(key);
-      var newState = reducer(oldState, action);
-      dirty = dirty || (oldState !== newState)
+
+    let finalState = state;
+    finalReducers.forEach( ( reducer, key ) => {
+      const oldState = state.get( key );
+      const newState = reducer( oldState, action );
+
       if (typeof newState === 'undefined') {
         throw new Error(getErrorMessage(key, action));
       }
-      return newState;
+
+      if( oldState != newState ) {
```

Comment:
Nice fix @phallguy. 

Since the Immutable `.set()` will only return a new instance if the value being set has changed you could probably remove this check. (The way it is might be more explicit though.)

If that is done, holding an explicit `oldState` reference becomes unnecessary and the `get()` on line 106 could be moved into line 107. (But that's just semantics.)

Just thought I'd share what I noticed.